pkg/solana/client/multinode/client_test.go

```diff
@@ -0,0 +1,152 @@
+package client
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client/multinode/config"
+)
+
+type TestRPC struct {
+	latestBlock int64
+}
+
+type TestHead struct {
+	blockNumber int64
+}
+
+func (t *TestHead) BlockNumber() int64        { return t.blockNumber }
+func (t *TestHead) BlockDifficulty() *big.Int { return nil }
+func (t *TestHead) IsValid() bool             { return true }
+
+func LatestBlock(ctx context.Context, rpc *TestRPC) (*TestHead, error) {
+	rpc.latestBlock++
+	return &TestHead{rpc.latestBlock}, nil
+}
+
+func initializeMultiNodeClient(t *testing.T) *MultiNodeClient[TestRPC, *TestHead] {
+	requestTimeout := 5 * time.Second
+	lggr := logger.Test(t)
+	cfg := &config.MultiNodeConfig{}
+	cfg.SetDefaults()
+	enabled := true
+	cfg.MultiNode.Enabled = &enabled
+
+	c, err := NewMultiNodeClient[TestRPC, *TestHead](cfg, &TestRPC{}, requestTimeout, lggr, LatestBlock, LatestBlock)
+	require.NoError(t, err)
+	return c
+}
+
+func TestMultiNodeClient_LatestBlock(t *testing.T) {
+	c := initializeMultiNodeClient(t)
+
+	t.Run("LatestBlock", func(t *testing.T) {
+		head, err := c.LatestBlock(tests.Context(t))
+		require.NoError(t, err)
+		require.Equal(t, true, head.IsValid())
+	})
+
+	t.Run("LatestFinalizedBlock", func(t *testing.T) {
+		finalizedHead, err := c.LatestFinalizedBlock(tests.Context(t))
+		require.NoError(t, err)
+		require.Equal(t, true, finalizedHead.IsValid())
+	})
+}
+
+func TestMultiNodeClient_HeadSubscriptions(t *testing.T) {
+	c := initializeMultiNodeClient(t)
```

Comment:
do not share client. Two tests spawn two pollers that concurrently write the latestBlock